Neos.Cache/Classes/Backend/FileBackend.php

```diff
@@ -268,16 +248,14 @@ public function findIdentifiersByTag($searchedTag)
             }
 
             $cacheEntryPathAndFilename = $directoryIterator->getPathname();
-            $lock = new Lock($cacheEntryPathAndFilename, false);
-            $index = (integer)file_get_contents($cacheEntryPathAndFilename, null, null, filesize($cacheEntryPathAndFilename) - self::DATASIZE_DIGITS, self::DATASIZE_DIGITS);
-            $metaData = file_get_contents($cacheEntryPathAndFilename, null, null, $index);
-            $lock->release();
-
-            $expiryTime = (integer)substr($metaData, 0, self::EXPIRYTIME_LENGTH);
+            $fileSize = filesize($cacheEntryPathAndFilename);
+            $index = (integer)$this->readCacheFile($cacheEntryPathAndFilename, $fileSize - self::DATASIZE_DIGITS, self::DATASIZE_DIGITS);
+            $metaData = $this->readCacheFile($cacheEntryPathAndFilename, $index, $fileSize - $index - self::DATASIZE_DIGITS);
+            $expiryTime = (integer)substr($metaData, -self::EXPIRYTIME_LENGTH, self::EXPIRYTIME_LENGTH);
             if ($expiryTime !== 0 && $expiryTime < $now) {
                 continue;
             }
-            if (in_array($searchedTag, explode(' ', substr($metaData, self::EXPIRYTIME_LENGTH, -self::DATASIZE_DIGITS)))) {
+            if (in_array($searchedTag, explode(' ', substr($metaData, 0, -self::EXPIRYTIME_LENGTH)))) {
```

Comment:
I basically just moved the expiry time to the end of the metadata block. That allows for quicker expiry checks, because it's no longer necessary to read the data size first to find the start of the arbitrary sized metadata. You can just read from the end of the file -EXPIRYTIME_LENGTH bytes and check.